dvc/dependency/repo.py

```diff
@@ -71,7 +72,20 @@ def fetch(self):
 
             out = repo.find_out_by_relpath(self.def_path)
             with repo.state:
-                repo.cloud.pull(out.get_used_cache())
+                try:
+                    repo.cloud.pull(out.get_used_cache())
+                except NoRemoteError:
+                    # Check if the file is available in the cache
+                    # If available in the cache then don't check out
+                    # else just raise the exception
+                    from dvc.repo import Repo
+
+                    source_repo = Repo()
```

Comment:
This will create a repo from cwd, which might not be in the repo at all. You are probably looking for `self.repo`.

Reply:
Yah actually I was looking for the same. I have modified the pr to use `self.repo`.